src/test/java/com/artipie/docker/DockerProxyIT.java

```diff
@@ -41,75 +37,70 @@
-    /**
-     * Docker client.
-     */
-    private DockerClient client;
-
-    /**
-     * Tested Artipie server.
-     */
-    private ArtipieServer server;
-
-    /**
-     * Docker repository.
-     */
-    private String repository;
+    @RegisterExtension
+    final TestDeployment deployment = new TestDeployment(
+        () -> TestDeployment.ArtipieContainer.defaultDefinition()
+            .withRepoConfig("docker/docker-proxy.yml", "my-docker"),
+        () -> new TestDeployment.ClientContainer("alpine:3.11")
+            .withPrivilegedMode(true)
+            .withWorkingDirectory("/w")
+    );
 
     @BeforeEach
-    void setUp(@TempDir final Path root) throws Exception {
-        this.server = new ArtipieServer(
-            root, "my-docker",
-            new RepoConfigYaml("docker-proxy").withRemotes(
-                Yaml.createYamlSequenceBuilder()
-                    .add(
-                        Yaml.createYamlMappingBuilder()
-                            .add("url", "registry-1.docker.io")
-                            .build()
-                    )
-                    .add(
-                        Yaml.createYamlMappingBuilder()
-                            .add("url", "mcr.microsoft.com")
-                            .build()
-                    )
-            )
-        );
-        final int port = this.server.start();
-        this.repository = String.format("localhost:%d", port);
-        this.image = new Image.ForOs();
-        final ArtipieServer.User user = ArtipieServer.ALICE;
-        this.client.login(user.name(), user.password(), this.repository);
-    }
-
-    @AfterEach
-    void tearDown() {
-        this.server.stop();
+    void setUp() throws Exception {
+        this.deployment.setUpForDockerTests();
     }
 
     @Test
-    void shouldPullRemote() throws Exception {
+    void shouldPullRemote() {
+        final Image image = new Image.ForOs();
         final String img = new Image.From(
-            this.repository,
-            String.format("my-docker/%s", this.image.name()),
-            this.image.digest(),
-            this.image.layer()
+            "artipie:8080",
+            String.format("my-docker/%s", image.name()),
+            image.digest(),
+            image.layer()
         ).remoteByDigest();
-        final String output = this.client.run("pull", img);
-        MatcherAssert.assertThat(
-            output,
-            new StringContains(String.format("Status: Downloaded newer image for %s", img))
+        List.of(
+            Pair.of(
+                "Failed to login to Artipie",
+                List.of(
+                    "docker", "login",
+                    "--username", "alice",
+                    "--password", "123",
+                    "artipie:8080"
+                )
+            ),
+            Pair.of(
+                "Failed to pull image",
+                List.of(
+                    "docker", "pull", img
+                )
+            )
+        ).forEach(
```

Comment:
There are several consernes:
- it's not easy to read and understand what is going on here with such approach. The best tests are plain, simple and obvious, with no complications (as each complication is a potential place for error, and each error in test can lead to false positive/negative affect). Thus, I'd suggest to get rid of the list and just call `this.deployment.assertExec()` twice with different arguments. I know that all docker tests are written this way, but do not think this is a good approach.
- exec code 0 does not actually mean the success here - are we sure the image was downloaded from artipie? I understand, that it was demanded in pull command, but we should consider the possibility that command was written incorrectly or for some reason it did not work and image was loaded from global registry. Thus, if possible, I'd suggest to add some expected strings into matcher (check conda test as an example). Also, some package managers return 0 exit code even package was not found